app/views/course/_suggested.html.erb

```diff
@@ -1,9 +1,43 @@
-<div class="tab-pane fade" id="next-section" role="tabpanel" aria-labelledby="next-section-tab">
-  <h2>Check Out This Section Next</h2>
-  <ul>
-    <% suggested = current_user.course_status.missing_sections.first %>
-    <li>
-      <%= link_to suggested.title, course_section_path(suggested) %>
-    </li>
-  </ul>
-</div>
\ No newline at end of file
+<div>
+ <h2 class="h3">Course Progress</h2>
+<hr>
+
+<div class="mb-4">
+  <div class="mb-3">
+    <p>You've completed <%= number_to_percentage(current_user.course_status.completion_rate * 100, precision: 0) %> of the sections in the course.</p>
+    <%= render "status/progress", status: current_user.course_status %>
+  </div>
+</div>
+
+<% last_quiz = current_user.course_status.next_in_line.last_section %>
+<% next_section = current_user.course_status.next_in_line.next_from_top %>
+<% last_quiz_score = last_quiz.answers.correct.size.fdiv(last_quiz.questions.size) %>
```

Comment:
In general I see this as a code smell, and whenever I find myself doing it, it probably means I need a partial.

Think about it, if you're saving `last_quiz` in order to use use multiple times on a given section, it's probably better to have this section as `last_quiz` partial and pass it as something like this instead:
```erb
<%= render "last_quiz", last_quiz: current_user.course_status.next_in_line.last_section %>
```

In this case, tho. I think a `next_in_line` partial might feel better.

Reply:
@caioertai I went with the next_in_line partial. Let me know if it looks better to you!